//Qt界面文件
<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Mymap</class>
 <widget class="QMainWindow" name="Mymap">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>631</width>
    <height>379</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Mymap</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QLabel" name="label_title">
    <property name="geometry">
     <rect>
      <x>150</x>
      <y>0</y>
      <width>311</width>
      <height>51</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Nirmala UI</family>
      <pointsize>20</pointsize>
      <weight>75</weight>
      <bold>true</bold>
     </font>
    </property>
    <property name="text">
     <string>Figure out Map Rodes !</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_inputn">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>60</y>
      <width>201</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Plz select n for map size (n*n):</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_rode">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>240</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Find rodes</string>
    </property>
   </widget>
   <widget class="QPushButton" name="pushButton_clear">
    <property name="geometry">
     <rect>
      <x>410</x>
      <y>280</y>
      <width>75</width>
      <height>23</height>
     </rect>
    </property>
    <property name="text">
     <string>Clear all</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_inputmap">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>90</y>
      <width>261</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Plz input an integer on each line for map  </string>
    </property>
   </widget>
   <widget class="QTextEdit" name="textEdit_map">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>170</y>
      <width>191</width>
      <height>141</height>
     </rect>
    </property>
    <property name="html">
     <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'SimSun'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_x1">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>100</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_x1">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>100</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Plz input x1 for start point(x1,y1):</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_y1">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>130</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Plz input y1 for start point(x1,y1):</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_y1">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>130</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_x2">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>160</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Plz input x2 for end point(x2,y2):</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_x2">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>160</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLineEdit" name="lineEdit_y2">
    <property name="geometry">
     <rect>
      <x>550</x>
      <y>190</y>
      <width>31</width>
      <height>20</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="QLabel" name="label_y2">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>190</y>
      <width>221</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>Plz input y2 for end point(x2,y2):</string>
    </property>
   </widget>
   <widget class="QComboBox" name="comboBox_n">
    <property name="geometry">
     <rect>
      <x>230</x>
      <y>60</y>
      <width>51</width>
      <height>22</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string extracomment="2*2">2</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>3</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>4</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>5</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>6</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>7</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>8</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>9</string>
     </property>
    </item>
   </widget>
   <widget class="QLabel" name="label_inputmap_2">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>110</y>
      <width>261</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>points: (Plz note: The larger, the harder </string>
    </property>
   </widget>
   <widget class="QLabel" name="label_inputmap_3">
    <property name="geometry">
     <rect>
      <x>30</x>
      <y>130</y>
      <width>271</width>
      <height>31</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="text">
     <string>to access. 0 stands for unaccessible)</string>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="geometry">
     <rect>
      <x>320</x>
      <y>70</y>
      <width>201</width>
      <height>16</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <underline>false</underline>
     </font>
    </property>
    <property name="text">
     <string>Note: Points from (1,1) to (n,n)</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>631</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuinput_your_map">
    <property name="title">
     <string>Help</string>
    </property>
    <addaction name="actionHelp"/>
   </widget>
   <addaction name="menuinput_your_map"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="actionFind_rodes">
   <property name="text">
    <string>Find rodes</string>
   </property>
  </action>
  <action name="actionClear_all">
   <property name="text">
    <string>Clear all</string>
   </property>
  </action>
  <action name="actionClear_start_point">
   <property name="text">
    <string>Clear start point</string>
   </property>
  </action>
  <action name="actionClear_End_point">
   <property name="text">
    <string>Clear End point</string>
   </property>
  </action>
  <action name="actionHelp">
   <property name="text">
    <string>Help</string>
   </property>
   <property name="menuRole">
    <enum>QAction::AboutRole</enum>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections>
  <connection>
   <sender>pushButton_clear</sender>
   <signal>clicked()</signal>
   <receiver>Mymap</receiver>
   <slot>clickClearButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>277</y>
    </hint>
    <hint type="destinationlabel">
     <x>401</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_rode</sender>
   <signal>clicked()</signal>
   <receiver>Mymap</receiver>
   <slot>clickRodeButton()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>430</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>409</x>
     <y>206</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <signal>clickExitButton()</signal>
  <slot>clickCloseButton()</slot>
  <slot>clickClearButton()</slot>
  <slot>clickRodeButton()</slot>
 </slots>
</ui>
